app/components/Settings/Preview/code-editor.jsx

```diff
@@ -0,0 +1,232 @@
+
+        dialect: PropTypes.string,
+        runQuery: PropTypes.func,
+        schemaRequest: PropTypes.object,
+        isLoading: PropTypes.bool
+    }
+
+
+    /**
+     * CodeEditor displays a CodeMirror editor to run SQL queries
+     *
+     * @param {object} props - Component properties
+     *
+     * @param {string}   props.value    - Value
+     * @param {onChange} props.onChange - Callback to handle a value change
+     *
+     * @param {string}   props.dialect       - Connector dialect
+     * @param {runQuery} props.runQuery      - Callback to run a SQL query
+     * @param {object}   props.schemaRequest - /connections/:id/sql-schemas API request
+     * @param {object}   props.isLoading     - true if last run query is still running
+     */
+    constructor(props) {
+        super(props);
+
+        /**
+         * @member {object} state - Component state
+         * @property {object} state.tables - Tables schemas (used in hintOptions)
+         * @property {number} state.height - Height in pixels
+         * @property {?number} state.width - Width in pixels
+         */
+        this.state = {
+            height: 250,
+            width: null,
+            minConstraints: [200, 74],
+            maxConstraints: [Infinity, Infinity],
+            tables: CodeEditor.computeAutocompleteTables(this.props)
+        };
+
+        this.onBeforeChange = (editor, data, value) => {
+            this.props.onChange(value);
+        };
+
+        this.onChange = (editor) => {
+            // Don't show autocomplete after a space or semicolon
+            const ch = editor.getTokenAt(editor.getCursor()).string.slice(-1);
+            if (!ch || ch === ' ' || ch === ';') {
+                return;
+            }
+
+            // hint options for specific plugin & general show-hint
+            // 'tables' is sql-hint specific
+            // 'disableKeywords' is also sql-hint specific, and undocumented but referenced in sql-hint plugin
+            // Other general hint config, like 'completeSingle' and 'completeOnSingleClick'
+            // should be specified here and will be honored
+            const hintOptions = {
+                tables: this.state.tables,
+                completeSingle: false,
+                completeOnSingleClick: true
+            };
+
+            // editor.hint.sql is defined when importing codemirror/addon/hint/sql-hint
+            // (this is mentioned in codemirror addon documentation)
+            // Reference the hint function imported here when including other hint addons
+            // or supply your own
+            CM.showHint(editor, CM.hint.sql, hintOptions);
+        };
+
+        this.editorDidMount = (editor) => {
+            this.editor = editor;
+
+            const wrapperElement = this.editor.getWrapperElement();
+            const height = wrapperElement.clientHeight;
+            const width = wrapperElement.clientWidth;
+
+            const minConstraints = [width, 74];
```

Comment:
Maybe put 74 as a constant at top with what it represents.  const MIN_WIDTH_CONSTRAINTS = 74